internal/server/create_investigation.go

```diff
@@ -0,0 +1,70 @@
+package server
+
+import (
+	"github.com/ministryofjustice/opg-go-common/template"
+	"github.com/ministryofjustice/opg-sirius-lpa-frontend/internal/sirius"
+	"net/http"
+	"strconv"
+)
+
+type CreateInvestigationClient interface {
+	CreateInvestigation(ctx sirius.Context, caseID int, caseType sirius.CaseType, investigation sirius.Investigation) error
+	Case(ctx sirius.Context, id int) (sirius.Case, error)
+}
+
+type createInvestigationData struct {
+	XSRFToken     string
+	Success       bool
+	Error         sirius.ValidationError
+	Case          sirius.Case
+	Investigation sirius.Investigation
+}
+
+func CreateInvestigation(client CreateInvestigationClient, tmpl template.Template) Handler {
+	return func(w http.ResponseWriter, r *http.Request) error {
+		caseID, err := strconv.Atoi(r.FormValue("id"))
+		if err != nil {
+			return err
+		}
+
+		caseType, err := sirius.ParseCaseType(r.FormValue("case"))
+		if err != nil {
+			return err
+		}
```

Comment:
`caseType` is only used for post requests so this can be moved to the if branch on line 47

Reply:
True, however it ensures the URL contains all the necessary params for when the POST is submitted. Otherwise it could be confusing to users if they entered the form data and an error appeared on submit